xbmc/windowing/XBMC_events.h

```diff
@@ -128,5 +128,6 @@ typedef struct XBMC_Event {
     XBMC_ButtonEvent keybutton;
     XBMC_SetFocusEvent focus;
   };
+  XBMC_Event() : type(), touch() {}
```

Comment:
You should certainly document that the largest member needs to be initialized, but I'm not sure I like this (how do we know what padding the compiler uses?).

Is it possible to name the union, and then do a memset inside the constructor? I think a memset is required to handle a union, but we can at least isolate it in the constructor.

Reply:
According to current c ++ standard it seems that all padding bytes is initialized to zero. Then final result should be same that using memset:

https://en.cppreference.com/w/cpp/language/zero_initialization

> If T is a union type, the first non-static named data member is zero-initialized and all padding is initialized to zero bits.

This answer on stackoverflow seems indicate that this way is correct:
https://stackoverflow.com/questions/11812555/how-to-zero-initialize-an-union

Reply:
Thanks for the SO link. The second answer says:

> In this case, I picked the largest member of your union since it's transparent. Otherwise you could explicitly create the union itself.

My concern is that the size isn't transparent. What does it mean about explicitly creating the union itself?

Reply:
> What does it mean about explicitly creating the union itself?

The truth is that I do not understand this phrase either.

Not being able to be sure that the code behaves the same in all compilers (padding bytes, etc.) I have removed this commit and included it like any other normal case of replacing `memset(x, 0, sizeof (x))` by `{}`. Then moved to "Commit 3".